default.nix

```diff
@@ -1,23 +1,26 @@
-{ pkgs ? (import <nixpkgs> {}), coq-version ? "master" }:
+{ pkgs ? (import <nixpkgs> {}), coq-version-or-url, shell ? false }:
 
 let
- coq =
-   let coq-version-parts = builtins.match "([0-9]+).([0-9]+)" coq-version; in
-   if coq-version-parts == null then
-     import (fetchTarball "https://github.com/coq/coq/tarball/${coq-version}") {}
-   else
-     pkgs."coq_${builtins.concatStringsSep "_" coq-version-parts}";
+  coq-version-parts = builtins.match "([0-9]+).([0-9]+)" coq-version-or-url;
+  coqPackages =
+    if coq-version-parts == null then
+      pkgs.mkCoqPackages (import (fetchTarball coq-version-or-url) {})
+    else
+      pkgs."coqPackages_${builtins.concatStringsSep "_" coq-version-parts}";
 in
 
+with coqPackages;
+
 pkgs.stdenv.mkDerivation {
 
-  name = "aac_tactics";
+  name = "aac-tactics";
+
+  buildInputs = with coq.ocamlPackages; [ ocaml findlib ]
+    ++ pkgs.lib.optionals shell [ merlin ocp-indent ocp-index ];
 
-  buildInputs = with coq.ocamlPackages;
-    [ coq ocaml findlib camlp5_strict ]
-    ++ pkgs.lib.optionals pkgs.lib.inNixShell [ merlin ocp-indent ocp-index ];
+  propagatedBuildInputs = [ coq ];
```

Comment:
Why is `coq` propagated?

Reply:
Thanks for asking. The reason is that if people want to *use* this package to run aac-tactics (with Coq master), it could be difficult to load the same version of Coq as well. And anyways, aac-tactics is useless without Coq, so it makes sense to propagate it, right? (As opposed to the case of an extracted program or CompCert.)

Reply:
Usually, libraries do not propagate the specific compiler used to build them, as said libraries might be used with a different compiler.

Reply:
>as said libraries might be used with a different compiler

Do you mean a different versions or a different implementation? In both cases, it doesn't seem like it is really relevant in the case of Coq since there is a single implementation and Coq is not really designed for stability of `vo` generated files. Even within the same version compiled with a different version of OCaml :confused: